### PR TITLE
pythonify this stuff

### PR DESCRIPTION
**Pull request by PaulSonOfLars**
make the code some level of sane. run it with python3 -m bot (and whatever args you want).
Kept the overall structure because it's still your bot, and refactoring is probably something you want to do

### CODE REVIEW

File: bot/__main__.py [hunk 1]
```diff
@@ -27,33 +27,29 @@
...
 webhook_listen = os.environ['NOLIFER_WEBHOOK_LISTEN'] \
-                                    if "NOLIFER_WEBHOOK_LISTEN" in os.environ \
-                                    else '127.0.0.1'
-webhook_url = "https://%s/%s" % \
-                            (os.environ['NOLIFER_WEBHOOK_BASEURL'] \
-                                if "NOLIFER_WEBHOOK_BASEURL" in os.environ \
-                                else webhook_listen, webhook_url_path)
+    if "NOLIFER_WEBHOOK_LISTEN" in os.environ \
+    else '127.0.0.1'
+webhook_url = "https://%s/%s" % (os.environ['NOLIFER_WEBHOOK_BASEURL'] if "NOLIFER_WEBHOOK_BASEURL" in os.environ
```
Comment (xdevs23) [inline bot/__main__.py:50]: 79-char limit!

File: bot/__main__.py [hunk 2]
```diff
@@ -112,6 +106,7 @@ def start_bot():
     updater.idle()
     print("Stopped")
 
+
```
Comment (xdevs23) [inline bot/__main__.py:109]: Huh?

File: bot/commands.py [hunk 3]
```diff
@@ -189,44 +190,43 @@ def launch_build(bot, update):
...
         else:
             update.message.reply_text("Cannot launch build, error code %i",
-                                        result_)
+                                      result_)
+
 
 def restart_bot(bot, update):
-    if update.message.chat_id == -1001068076699 or \
-       update.message.chat_id == 11814515:
+    if update.message.chat_id == -1001068076699 or update.message.chat_id == 11814515:
```
Comment (xdevs23) [inline bot/commands.py:217]: 79-char limit!

File: bot/custom_filters.py [hunk 4]
```diff
@@ -23,40 +23,37 @@
...
 _latest_build_file = getenviron(
     "NOLIFER_LATEST_BUILD_FILE",
     "/var/lib/jenkins/workspace/halogenOS/%s-latest.txt")
 
+
 class HashMessageFilter(BaseFilter):
     def filter(self, message):
-        return message.text != None and \
-                len(message.text) >= 2 and message.text[0] == '#'
+        return message.text is not None and len(message.text) >= 2 and message.text[0] == '#'
```
Comment (xdevs23) [inline bot/custom_filters.py:35]: 79-char limit!

File: bot/custom_filters.py [hunk 5]
```diff
@@ -23,40 +23,37 @@
...
+
 
 def on_hash_message(bot, update):
     msg_split = update.message.text.split()
     hashtag_item = msg_split[0][1:]
-    if update.message.chat_id == -1001068076699 or \
-         update.message.chat_id == 11814515:
+    if update.message.chat_id == -1001068076699 or update.message.chat_id == 11814515:
```
Comment (xdevs23) [inline bot/custom_filters.py:41]: 79-char limit!

File: bot/custom_filters.py [hunk 6]
```diff
@@ -23,40 +23,37 @@
...
+                                re.match('^[\w-]+$', msg_split[1]) is None:
                     update.message.reply_text("Nice try ;)")
                 else:
-                    file = open(
-                        _latest_build_file \
-                            % (msg_split[1] \
-                                if len(msg_split) > 1 else "oneplus2"), "r")
-                    update.message.reply_text(file.read())
-                    file.close()
+                    with open(_latest_build_file % (msg_split[1] if len(msg_split) > 1 else "oneplus2"), "r") as f:
```
Comment (xdevs23) [inline bot/custom_filters.py:48]: 79-char limit!

File: bot/utils.py [hunk 7]
```diff
@@ -1,5 +1,5 @@
-
```
Comment (xdevs23) [inline bot/utils.py:1]: Useless changes in this file.

File: bot/commands.py [hunk 8]
```diff
@@ -189,44 +190,43 @@ def launch_build(bot, update):
...
-        tmpfile.flush()
-        tmpfile.close()
+        with open("/tmp/nolifer-stop-reason", "w") as tmpfile:
+            tmpfile.write("restart %s" % update.message.chat_id)
         # Send SIGTERM to terminate normally
         os.kill(os.getpid(), signal.SIGTERM)
     else:
         update.message.reply_text("Sorry, you are not allowed to do that here")
 
+
```
Comment (xdevs23) [inline bot/commands.py:226]: uh huh

File: bot/commands.py [hunk 9]
```diff
@@ -18,24 +18,23 @@
...
 # Python imports
 import os
-import sys
 import requests
 import signal
 from subprocess import call
 from os.path import expanduser
 
-# Library imports
-import telegram
```
Comment (xdevs23) [inline bot/commands.py:28]: Are you sure?

**Review by xdevs23 — CHANGES_REQUESTED** [review]
Not sure if your concept of using `__init__` and that stuff is what I like.

You added alot of empty lines. Why?

Why do you change the indentation of many lines even though they are ok? Do you have a reference with clear rules on how to do it?

**Comment by PaulSonOfLars** [comment]
Yes, I do have whitespace rules, it's called pep8
Yes, those newlines are there because of pep8
Yes, pep8 doesn't use an 80 char limit, but rather 100 or 120, depending.

**Comment by xdevs23** [comment]
> Yes, I do have whitespace rules, it's called pep8

Show me the exact place


> Yes, pep8 doesn't use an 80 char limit, but rather 100 or 120, depending.

PEP8 prefers 79: https://www.python.org/dev/peps/pep-0008/#id19

> Limit all lines to a maximum of 79 characters.
>
> ...
> 
> Some teams strongly prefer a longer line length. For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the nominal line length from 80 to 100 characters (effectively increasing the maximum length to 99 characters), provided that comments and docstrings are still wrapped at 72 characters.

It's a preference.

**Pull request closed without merging**